src/main/java/model/Transaction.java

```diff
@@ -0,0 +1,42 @@
+package model;
+
+import service.SeparatorService;
+import service.impl.SeparatorServiceImpl;
+import strategy.OperationHandler;
+
+public class Transaction {
+    private int value;
+    private String fruit;
+    private OperationHandler operationHandler;
+
+    public void createTransaction(String line) {
+        SeparatorService separator = new SeparatorServiceImpl(line);
+        setFruitOperation(separator.getOperationFromLine());
+        setFruit(separator.getFruitFromLine());
+        setValue(separator.getValueFromLine());
+    }

```

Comment:
move it from the model to TransactionCreatorService. Also, you should avoid dependencies between services.